core/src/test/java/com/microsoft/applicationinsights/internal/jmx/JmxDataFetcherTest.java

```diff
@@ -96,15 +101,20 @@ private static void performTest(
             List<JmxAttributeData> attributes,
             double expectedInt,
             double expectedDouble,
-            double expectedLong) throws Exception {
-        Map<String, Collection<Object>> result = JmxDataFetcher.fetch("JSDKTests:type=TestStub", attributes);
+            double expectedLong) {
+        try {
+            Map<String, Collection<Object>> result = JmxDataFetcher.fetch("JSDKTests:type=TestStub", attributes);
 
-        assertNotNull(result);
-        assertEquals(3, result.size());
+            assertNotNull(result);
+            assertEquals(3, result.size());
 
-        verify(result, "Int", expectedInt);
-        verify(result, "Double", expectedDouble);
-        verify(result, "Long", expectedLong);
+            verify(result, "Int", expectedInt);
+            verify(result, "Double", expectedDouble);
+            verify(result, "Long", expectedLong);
+        }
+        catch(Exception e) {
+            Assert.fail("Exception thrown: "+e);
+        }
```

Comment:
it's not clear why this catch is needed?

Reply:
Not very familiar with the required test structure. Do we need a way to handle the exception here?